storage/client/queries/queries.go

```diff
@@ -404,10 +404,29 @@ const (
 			($4::text IS NULL OR evs.tx_hash = $4::text OR evs.tx_eth_hash = $4::text) AND
 			($5::text IS NULL OR evs.type = $5::text) AND
 			($6::bytea IS NULL OR evs.evm_log_signature = $6::bytea) AND
-			($7::text IS NULL OR evs.related_accounts @> ARRAY[$7::text])
+			($7::text IS NULL OR evs.related_accounts @> ARRAY[$7::text]) AND
+			($8::text IS NULL OR (
+			    -- Currently this only supports EVM smart contracts.
+			    evs.type = 'evm.log' AND
+			    evs.body ->> 'address' = (
+			        SELECT encode(address_data, 'base64')
+			        FROM chain.address_preimages
+			        WHERE
+			            address = $8::text AND
+						context_identifier = 'oasis-runtime-sdk/address: secp256k1eth' AND
+						context_version = 0
+				)
```

Comment:
An equivalent expression is `encode(eth_preimage($8::oasis_addr), 'base64')`, via [this func](https://github.com/oasisprotocol/nexus/blob/f495dac742ed91fd6e32336dfbc99d00955c9abd/storage/migrations/18_refactor_dead_reckoning.up.sql#L8-L25)

Reply:
o thanks

Reply:
switched to that